File: java/factory/MasterCard.java
package factory;

public class MasterCard extends InternationalPaymentMethod {
    public MasterCard(String cardHolder, String cardNumber, String cvv, String expiryDate) {
        super(cardHolder, cardNumber, cvv, expiryDate);
    }

    @Override
    public void authorizePayment() {
        System.out.println("Authorizing master card....");
    }

    @Override
    public void startMoneyTransfer() {
        System.out.println("Start money transferring for a master card....");
    }

    @Override
    public void calculatePaymentFees(double amount) {
        System.out.println("Calculating master card payment fees....");
    }
}

File: typescript/factory/MasterCard.ts
import { InternationalPaymentMethod } from "./InternationalPaymentMethod";

export class MasterCard extends InternationalPaymentMethod {
    constructor(
      cardHolder: string,
      cardNumber: string,
      cvv: string,
      expiryDate: string
    ) {
      super(cardHolder, cardNumber, cvv, expiryDate);
    }
  
    authorizePayment(): void {
      console.log("Authorizing master card....");
    }
  
    startMoneyTransfer(): void {
      console.log("Start money transferring for a master card....");
    }
  
    calculatePaymentFees(amount: number): void {
      console.log("Calculating master card payment fees....");
    }
  }